js/apps/system/_admin/aardvark/APP/react/src/views/query/queryResults/useDisplayTypes.tsx

```diff
@@ -0,0 +1,294 @@
+import L from "leaflet";
+import { GeodesicLine } from "leaflet.geodesic";
+import React from "react";
+import { QueryResultType } from "../ArangoQuery.types";
+
+export type DisplayType = "json" | "graph" | "table" | "geo";
+type GraphDataType = "graphObject" | "edgeArray";
+/**
+ * check if result could be displayed as graph
+ * case a) result has keys named vertices and edges
+ * case b) 95% results _from and _to attribute
+ */
+
+export const useDisplayTypes = ({
+  queryResult
+}: {
+  queryResult: QueryResultType;
+}) => {
+  const [displayTypes, setDisplayTypes] = React.useState<Array<DisplayType>>(
+    []
+  );
+  const [currentDisplayType, setCurrentDisplayType] =
+    React.useState<DisplayType>("json");
+  const [graphDataType, setGraphDataType] =
+    React.useState<GraphDataType>("graphObject");
+  React.useEffect(() => {
+    const { isGraph, graphDataType } = detectGraph({
+      result: queryResult.result
+    });
+    if (isGraph && graphDataType) {
+      setDisplayTypes(["graph"]);
+      setGraphDataType(graphDataType);
+      setCurrentDisplayType(prevView => {
```

Comment:
```suggestion
      setCurrentDisplayType(prevView => (prevView !== "graph" ? "graph" : prevView))
```

Reply:
I don't see any benefit of using a ternary in this case, infact what it is currently seems cleaner and easier to read